platform-manageability-agent/cmd/platform-manageability-agent/platform-manageability-agent.go

```diff
@@ -99,8 +106,83 @@ func main() {
 
 	log.Info("Platform Manageability Agent started successfully")
 
+	tlsConfig, err := auth.GetAuthConfig(auth.GetAuthContext(ctx, confs.AccessTokenPath), nil)
+	if err != nil {
+		log.Fatalf("TLS configuration creation failed! Error: %v", err)
+	}
+
+	dmMgrClient := comms.ConnectToDMManager(auth.GetAuthContext(ctx, confs.AccessTokenPath), confs.Manageability.ServiceURL, tlsConfig)
+
+	var (
+		wg sync.WaitGroup
+
+		isAMTEnabled                = false
+		amtStatusCheckInterval      = 30 * time.Second // TODO: Make this configurable.
+		lastAMTStatusCheckTimestamp int64
+	)
+	go func() {
```

Comment:
```suggestion
	wg.Add(1)
	go func() {
```
For each go function created, we need to include a `wg.Add(1)` so the can be synchronized correctly